[SRVKS-1100] Add docs on debugging Serverless applications (#72687)
openshift/openshift-docs: modules/serverless-applications-checking-ingress-istio-routing.adoc

// Module included in the following assemblies:
//
// * knative-serving/debugging-serverless-applications.adoc

:_content-type: PROCEDURE
[id="serverless-applications-checking-ingress-istio-routing_{context}"]
= Checking Ingress and Istio routing

> Sometimes, the Ingress and Istio routing have issues. You can see the details on them using the `oc` tool.

Reviewer: Probably a hint here would be good to tell people that this is only relevant when istio is used as an ingress layer.

> Sometimes, when Istio is used as an Ingress layer, the Ingress and Istio routing have issues. You can see the details on them using the `oc` tool.

Reviewer 2: Per [term glossary](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#openshift-cli):
```suggestion
Sometimes, when Istio is used as an Ingress layer, the Ingress and Istio routing have issues. You can see the details on them by using the OpenShift CLI (`oc`).
```

.Procedure

> . List all Ingress resources and their corresponding labels:

Author: Applied where appropriate across whole PR.

+
[source,terminal]
----
$ oc get ingresses.networking.internal.knative.dev -o=custom-columns='NAME:.metadata.name,LABELS:.metadata.labels'
----
+
.Example output
[source,terminal]
----
NAME            LABELS
helloworld-go   map[serving.knative.dev/route:helloworld-go serving.knative.dev/routeNamespace:default serving.knative.dev/service:helloworld-go]
----
+
In this output, labels `serving.knative.dev/route` and `serving.knative.dev/routeNamespace` indicate the `Route` where the Ingress resource resides. Your `Route` and Ingress should be listed.
+

> If your Ingress does not exist, the route controller assumes that the `Revision` objects targeted by your `Route` or `Service` are not ready. Proceed to the following procedures sections to diagnose `Revision` readiness status.

Reviewer 2: "Proceed to the following procedures sections"

Considering modularized content and not knowing for sure where the module might appear in the future, Is there a way to say this without using a directional term? Maybe "Proceed with other debugging procedures" or similar?

> . If your Ingress is listed, examine the `ClusterIngress` object created for your `Route`:

Reviewer 2: ```suggestion
. If your Ingress is listed, examine the `ClusterIngress` object created for your route:
```

+
[source,terminal]
----
$ oc get ingresses.networking.internal.knative.dev <ingress_name> --output yaml
----
+
In the status section of the output, if the condition with `type=Ready` has the status of `True`, then Ingress is working correctly. Otherwise, the output contains error messages.

> . If Ingress has the status of `Ready`, then there is a corresponding `VirtualService`. Verify the configuration of `VirtualService`:

Reviewer 2: ```suggestion
. If Ingress has the status of `Ready`, then there is a corresponding `VirtualService` object. Verify the configuration of the `VirtualService` object:
```
